### DISCUSSION TITLE
HL7 Message ORC Segment GetQuantityTiming Part Cant Able to Add RepeatPattern Subelements Using NHapi.Model.V25

### DISCUSSION THREAD

**Discussion by Raseem001** [nHapiNET/nHapi#620] [Q&A] [answered]
Sir 
I need to Add Repeat Pattern Following Data For Malaffi Submission but the Code Showing Error of Following 

Csharp Code

```csharp
using NHapi.Model.V25.Datatype;
using NHapi.Model.V25.Message;
using NHapi.Model.V25.Segment;
using NHapi.Base.Model;  

var timing = orc.GetQuantityTiming(0);
timing.Quantity.Identifier.Value = "BID";
timing.Quantity.Text.Value = "twice a day at institution-specified times";
timing.Quantity.NameOfCodingSystem.Value = "MALAFFI";
```

```
CQ: does not Contain a Definition for 'Identifier' and no accessible extension method 'Identifier' accepting a first argument of type 'CQ' could be found(are you mssing a using directive or an assempbly reference ?)
```

RDS-O13 ORC Segment Sample Message : 

`ORC|NW|678998^^SYSTEMCODE|678998^^SYSTEMCODE||active||^QID^four times a day at institution-specified times^MALAFFI^4 Days^20191117120000^20191119180000^R^PRN SNOMED||20191113102719|GD18123^Test1^test4^^^^^^&SYSTEMCODE-DOHID|GD18123^Test1^test4^^^^^^&SYSTEMCODE-DOHID|GD18123^Test1^test4^^^^^^&SYSTEMCODE-DOHID`

**Comment by milkshakeuk** [comment]
The error you are seeing is accurate, according to the 2.5 specification the quantity (data type `TQ` Quantity/Timing) does not contain a field called `Identifier` see [here](https://hl7-definition.caristix.com/v2/HL7v2.5/DataTypes/TQ) for available fields.

However, looking at your example encoded ORC segment perhaps you are looking for something like this:

```csharp
var timing = orc.GetQuantityTiming(0);
timing.Interval.RepeatPattern.Value = "BID";
timing.Duration.Value = "twice a day at institution-specified times";
timing.StartDateTime.Time.Value = "20191117120000";
timing.EndDateTime.Time.Value = "20191119180000";
timing.Priority.Value = "R";
timing.OccurrenceDuration.NameOfCodingSystem.Value = "MALAFFI";
timing.OccurrenceDuration.Text.Value = "4 Days";
```

The output of encoded `TQ` Quantity/Timing (ORC-7) would look like this: 
`^BID^twice a day at institution-specified times^20191117120000^20191119180000^R^^^^^&4 Days&MALAFFI`
however I am not 100% sure based on your sample (ORC-7) which seems like it's not quite aligned with the HL7 standard.
`^QID^four times a day at institution-specified times^MALAFFI^4 Days^20191117120000^20191119180000^R^PRN SNOMED`

**Comment by milkshakeuk** [comment]
Unless you meant to assign the Quantity Units (ORC.7.1.2) which does seem to have those fields?

```csharp
using NHapi.Model.V25.Datatype;
using NHapi.Model.V25.Message;
using NHapi.Model.V25.Segment;
using NHapi.Base.Model;  

var timing = orc.GetQuantityTiming(0);
timing.Quantity.Units.Identifier.Value = "BID";
timing.Quantity.Units.Text.Value = "twice a day at institution-specified times";
timing.Quantity.Units.NameOfCodingSystem.Value = "MALAFFI";
timing.StartDateTime.Time.Value = "20191117120000";
timing.EndDateTime.Time.Value = "20191119180000";
timing.Priority.Value = "R";
```

but that produces this encoded TQ:
`&BID^^^20191117120000^20191119180000^R`

**Comment by Raseem001** [comment]
> ^QID^four times a day at institution-specified times^MALAFFI^4 Days^20191117120000^20191119180000^R^PRN SNOMED

in this Format Malaffi need to Get.  Format is given in their document is like this. After Repeat Pattern Description We need to Put MALAFFI in it.

**Comment by milkshakeuk** [comment, marked as answer]
If you need it exactly as per the sample, you would need to assign values that do not match the fields intended use case:

```csharp
var timing = orc.GetQuantityTiming(0);
timing.Interval.RepeatPattern.Value = "QID";
timing.Duration.Value = "four times a day at institution-specified times"; // is this a duration?
timing.StartDateTime.Time.Value = "MALAFFI"; // doesn't look like a Start Date Time?
timing.EndDateTime.Time.Value = "4 Days";    // again doesn't look like a DateTime
timing.Priority.Value = "20191117120000";    // assigning a date to priority?
timing.Condition.Value = "20191119180000";   // assigning a date to a condition?
timing.Text.Value = "R";                     // pretty sure this is not intended
timing.Conjunction.Value = "PRN SNOMED";     // same here
```

The output of encoded TQ Quantity/Timing (ORC-7) would look like this:
`^QID^four times a day at institution-specified times^MALAFFI^4 Days^20191117120000^20191119180000^R^PRN SNOMED`

**Comment by Raseem001** [comment]
Let me Check I think This Will Work Thank You Very Much

________________________________

**Reply by milkshakeuk** [reply]
If this worked, can you mark the specific response as the answer, thanks!
